test/Riok.Mapperly.Tests/Mapping/UseMapperTest.cs

```diff
@@ -0,0 +1,288 @@
+            partial B Map(A source);
+            private partial BExternal MapInternal(AExternal source);
+            """,
+            "record A(AExternal Value);",
+            "record B(BExternal Value);",
+            "record AExternal();",
+            "record BExternal();",
+            "class OtherMapper { public BExternal ToBExternal(AExternal source) => new BExternal(); }"
+        );
+        TestHelper
+            .GenerateMapper(source)
+            .Should()
+            .HaveMapMethodBody(
+                """
+                var target = new global::B(MapInternal(source.Value));
+                return target;
+                """
+            );
+    }
+
+    [Fact]
+    public void PreferInternalImplementedMapping()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            """
+            [UseMapper]
+            private readonly OtherMapper _otherMapper = new();
+            partial B Map(A source);
+            private BExternal MapInternal(AExternal source) => new BExternal();
+            """,
+            "record A(AExternal Value);",
+            "record B(BExternal Value);",
+            "record AExternal();",
+            "record BExternal();",
+            "class OtherMapper { public BExternal ToBExternal(AExternal source) => new BExternal(); }"
+        );
+        TestHelper
+            .GenerateMapper(source)
+            .Should()
+            .HaveMapMethodBody(
+                """
+                var target = new global::B(MapInternal(source.Value));
+                return target;
+                """
+            );
+    }
+
+    [Fact]
+    public void NullableFieldShouldDiagnostic()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            """
+            [UseMapper]
+            private readonly OtherMapper? _otherMapper;
+            partial B Map(A source);
+            """,
+            "record A(AExternal Value);",
+            "record B(BExternal Value);",
+            "record AExternal();",
+            "record BExternal();",
+            "class OtherMapper { public BExternal ToBExternal(AExternal source) => new BExternal(); }"
+        );
+        TestHelper
+            .GenerateMapper(source, TestHelperOptions.AllowDiagnostics)
+            .Should()
+            .HaveDiagnostic(
+                DiagnosticDescriptors.ExternalMapperMemberCannotBeNullable,
+                "The used mapper member Mapper._otherMapper cannot be nullable"
+            )
+            .HaveAssertedAllDiagnostics()
+            .HaveMapMethodBody(
+                """
+                var target = new global::B(MapToBExternal(source.Value));
+                return target;
+                """
+            );
+    }
+
+    [Fact]
+    public void DisabledNullableShouldWork()
+    {
+        var source = TestSourceBuilder.MapperWithBodyAndTypes(
+            """
+            [UseMapper]
+            private readonly OtherMapper _otherMapper;
```

Comment:
Could you add a test where the used mapper is a property instead of a field?

Reply:
Added, thanks for noting 😊